compaction/compaction_manager.hh

```diff
@@ -112,6 +116,12 @@ private:
         sstables::compaction_stopped_exception make_compaction_stopped_exception() const;
     };
 
+    struct major_compaction_task;
+    struct custom_compaction_task;
+    struct regular_compaction_task;
+    struct offstrategy_compaction_task;
+    struct rewrite_sstables_compaction_task;
+
```

Comment:
I don't remember exactly, I'll see if they can be removed after `class task` is made public in compaction_manager, so it can be inherited from by classes unrelated to compaction_manager`.

Reply:
It turns out that the tasks do need access to private compaction_manager members.
(E.g. `_maintenance_ops_sem`, `postponed_compactions_reevaluation()`, `reevaluate_postponed_compactions()`)